<launch>
    <arg name="nms_threshold" default="0.45" />
    <arg name="score_threshold" default="0.30" />

    <arg name="names_file" default="$(find vision_detector)/models/coco.names" />
    <arg name="network_definition_file" default="" /> <!-- For compatibility with vision_darknet_detect -->
    <arg name="pretrained_model_file" default="$(find vision_detector)/models/yolo_v2_tiny.pb" />

    <arg name="camera_id" default="/" />
    <arg name="image_src" default="/image_raw" />

    <node pkg="vision_detector" name="vision_detector" type="vision_detector" output="screen">
        <param name="nms_threshold" type="double" value="$(arg nms_threshold)"/>
        <param name="score_threshold" type="double" value="$(arg score_threshold)" />

        <param name="network_definition_file" type="str" value="$(arg network_definition_file)" />
        <param name="pretrained_model_file" type="str" value="$(arg pretrained_model_file)" />
        <param name="names_file" type="str" value="$(arg names_file)"/>

        <param name="image_raw_node" type="str" value="$(arg camera_id)$(arg image_src)"/>
    </node>

    <node pkg="detected_objects_visualizer" type="visualize_rects" name="yolo2tiny_rects" output="screen">
        <param name="image_src" value="/vision_detector/image_raw"/>
        <param name="image_out" value="/image_rects"/>
        <param name="object_src" value="/vision_detector/objects"/> <!-- this is fixed by definition -->
    </node>

</launch>